packages/react-slider/src/components/Slider/Slider.test.tsx

```diff
@@ -45,207 +45,201 @@ describe('Slider', () => {
+        [currentValue],
+      );
+
+      return <Slider value={currentValue} min={0} max={100} />;
     };
 
     render(<SliderTestComponent />);
-    expect(sliderRef.current.value).toEqual(0);
+
+    expect(sliderRef.current.getValue()).toEqual(10);
   });
 
   it('clamps an initial (defaultValue) that is out of bounds', () => {
-    let sliderRef: any;
-
-    const SliderTestComponent = () => {
-      sliderRef = React.useRef(null);
+    let internalValue;
 
-      return <Slider defaultValue={-10} min={0} max={100} ref={sliderRef} />;
+    const onChange = (incomingValue: number) => {
+      internalValue = incomingValue;
     };
 
-    render(<SliderTestComponent />);
-    expect(sliderRef.current.value).toEqual(0);
-  });
-
-  it('clamps an initial (value) that is out of bounds', () => {
-    let sliderRef: any;
-
-    const SliderTestComponent = () => {
-      sliderRef = React.useRef(null);
+    render(<Slider defaultValue={-10} min={0} max={100} onChange={onChange} data-testid="test" />);
+    const sliderRoot = screen.getByTestId('test');
 
-      return <Slider value={-10} min={0} max={100} ref={sliderRef} />;
-    };
+    fireEvent.keyDown(sliderRoot, { key: 'ArrowLeft' });
 
-    render(<SliderTestComponent />);
-    expect(sliderRef.current.value).toEqual(0);
+    expect(internalValue).toEqual(0);
   });
 
-  it('clamps provided controlled (value) that is out of bounds', () => {
-    let sliderRef: any;
-    let imperativeRef: any;
-
-    const SliderTestComponent = () => {
-      const [sliderValue, setSliderValue] = React.useState(-10);
-      sliderRef = React.useRef(null);
-      imperativeRef = React.useRef(null);
-
-      React.useImperativeHandle(imperativeRef, () => ({
-        getSliderValue: () => {
-          return sliderValue;
-        },
-      }));
-
-      const onChange = (value: number) => setSliderValue(value);
+  it('clamps an initial (value) that is out of bounds', () => {
+    let internalValue;
 
-      return <Slider value={sliderValue} min={0} max={100} onChange={onChange} ref={sliderRef} />;
+    const onChange = (incomingValue: number) => {
+      internalValue = incomingValue;
     };
 
-    render(<SliderTestComponent />);
+    render(<Slider value={-10} min={0} max={100} onChange={onChange} data-testid="test" />);
+    const sliderRoot = screen.getByTestId('test');
+
+    fireEvent.keyDown(sliderRoot, { key: 'ArrowLeft' });
 
-    expect(sliderRef.current.value).toEqual(0);
-    expect(imperativeRef.current.getSliderValue()).toEqual(0);
+    expect(internalValue).toEqual(0);
   });
 
-  it('calls (onChange) when dragged', () => {
-    let sliderRef: any;
-    const onChange = jest.fn();
+  it('calls (onChange) when pointerDown', () => {
+    let totalCalls = 0;
 
     const SliderTestComponent = () => {
-      sliderRef = React.useRef(null);
-      return <Slider defaultValue={5} onChange={onChange} ref={sliderRef} data-testid="test" />;
+      const onChange = (incomingValue: number) => {
```

Comment:
I don't think you need a wrapper component here. Also you can use [Jest mock functions](https://archive.jestjs.io/docs/en/24.x/mock-functions) to count calls:
```ts
const onChange = jest.fn();
// do stuff...
expect(onChange).toHaveBeenCalledTimes(1);
```
You can optionally also use them to check arguments in other tests:
```ts
const mockFn = jest.fn();
// If you want to test all the arguments
expect(mockFn).toHaveBeenLastCalledWith('a', 'b');
// Or to test only one argument (like if the other is an event) it's more awkward.
// First array access is the call index and second is the argument index.
expect(mockFn.mock.calls[0][0]).toBe('a');
```